test/automated-browser-tests.js

```diff
@@ -16,199 +16,56 @@
-      }
-    });
-  };
-
-  const queueUnitTest = (browserName, browserPath, seleniumBrowserID, options) => {
-    if (!browserPath) {
-      console.warn(`${browserName} path wasn\'t found so skipping`);
-      return;
-    }
-
-    it(`should pass all tests in ${browserName}`, () => {
-      globalDriverReference = new webdriver
-        .Builder()
-        .forBrowser(seleniumBrowserID)
-        .setChromeOptions(options)
-        .setFirefoxOptions(options)
-        .build();
+  const queueUnitTest = browserInfo => {
+    it(`should pass all tests in ${browserInfo.prettyName}`, () => {
+      globalDriverReference = browserInfo.getSeleniumDriver();
 
-      return performTests(browserName, globalDriverReference);
+      return automatedBrowserTesting.runMochaTests(
+        browserInfo.prettyName,
+        globalDriverReference,
+        `${testServerURL}/test/browser-tests/`
+      )
+      .then(testResults => {
+        automatedBrowserTesting.manageTestResults(testResults);
+      });
     });
   };
 
-  function configureBrowserTests() {
-    // Chrome Stable
-    try {
-      const CHROME_PATH = which.sync('google-chrome');
-      const chromeStableOpts = new chromeOptions.Options();
-      chromeStableOpts.setChromeBinaryPath(CHROME_PATH);
-      queueUnitTest('Chrome Stable', CHROME_PATH, 'chrome', chromeStableOpts);
-    } catch (err) {
-      console.error('Unable to find Chrome Stable.');
-    }
-
-
-    // Chrome Beta
-    try {
-      const CHROME_BETA_PATH = which.sync('google-chrome-beta');
-      const chromeBetaOpts = new chromeOptions.Options();
-      chromeBetaOpts.setChromeBinaryPath(CHROME_BETA_PATH);
-      queueUnitTest('Chrome Beta', CHROME_BETA_PATH, 'chrome', chromeBetaOpts);
-    } catch (err) {
-      console.error('Unable to find Chrome Beta.');
-    }
-
-    // Firefox Default Install
-    try {
-      const FIREFOX_PATH = which.sync('firefox');
-      const ffStableOpts = new firefoxOptions.Options();
-      ffStableOpts.setBinary(FIREFOX_PATH);
-      queueUnitTest('Firefox Stable', FIREFOX_PATH, 'firefox', ffStableOpts);
-    } catch (err) {
-      console.error('Unable to find default Firefox.');
-    }
-
-
-    // Firefox Beta in specific path on Travis
-    if (process.env.TRAVIS) {
-      try {
-        const FIREFOX_BETA_PATH_FOR_TRAVIS = './firefox/firefox';
-        const ffBetaOpts = new firefoxOptions.Options();
-        ffBetaOpts.setBinary(FIREFOX_BETA_PATH_FOR_TRAVIS);
-        queueUnitTest('Firefox Beta', FIREFOX_BETA_PATH_FOR_TRAVIS, 'firefox',
-          ffBetaOpts);
-      } catch (err) {
-        console.error('Unable to find Firefox beta.');
-      }
-    }
-  }
-
-  function clearTestDataBuild(done) {
-    del(GLOBAL.config.dest + '/**').then(() => done(), done);
-  }
-
-  describe('Test Dev Environment', function() {
-    configureBrowserTests();
+  const automatedBrowsers = automatedBrowserTesting.getAutomatedBrowsers();
+  automatedBrowsers.forEach(browserInfo => {
```

Comment:
Nit: You can put this all on one line as `automatedBrowsers.forEach(browserInfo => queueUnitTest(browserInfo));`